ui/src/components/credentials/LinkDetails.tsx

```diff
@@ -81,29 +81,33 @@ export function LinkDetails() {
               });
             } else {
               setCredentialSubjectValue({
-                error: `The type "${parsedCredentialSubject.data.type}" is not a valid type for the attribute "credentialSubject".`,
+                error: buildAppError(
```

Comment:
Wierd use the `buildAppError` inside the component, maybe better change the error type to string and extract the `message from the last case

Reply:
This `AsyncTask<ObjectAttributeValue, string>` is not linked to the refactor, this is the way that we use the error in this component, we are setting the error as we need it in some components, and we are accessing the message inside ZodError in other places. but it's fine. 
Maybe all that logic should be moved to a parser and cleaned the component.

Could you reuse it with the `CredentialDetails`? I'm seeing that is the same code. 
```
 if (parsedCredentialSubject.success) {
            if (parsedCredentialSubject.data.type === "object") {
              setCredentialSubjectValue({
                data: parsedCredentialSubject.data,
                status: "successful",
              });
            } else {
              setCredentialSubjectValue({
                error: buildAppError(
                  `The type "${parsedCredentialSubject.data.type}" is not a valid type for the attribute "credentialSubject".`
                ),
                status: "failed",
              });
            }
          } else {
            setCredentialSubjectValue({
              error: buildAppError(parsedCredentialSubject.error),
              status: "failed",
            });
          }
        } else {
          setCredentialSubjectValue({
            error: buildAppError(
              `Could not find the attribute "credentialSubject" in the object's schema.`
            ),
            status: "failed",
          });
        }
      } else {
        setCredentialSubjectValue({
          error: response.error,
          status: "failed",
        });
      }
    });
```

Reply:
A lot of comments...

> This `AsyncTask<ObjectAttributeValue, string>` is not linked to the refactor

Why not? This refactor applies to the error management of the whole app, this `AsyncTask<ObjectAttributeValue, string>` included.

> this is the way that we use the error in this component, we are setting the error as we need it in some components

Not sure I understand this. The way this component uses the errors has not changed. Only the shape of the error has, which now is required to be `AppError`.

> and we are accessing the message inside ZodError in other places

You're right there! I missed the usage of `processZodError` in the env context. Fixed. Now `processZodError` is not even exported.

> Maybe all that logic should be moved to a parser and cleaned the component.

Can you extend on this or provide a PoC, I'm not sure I'm following.

> Could you reuse it with the `CredentialDetails`? I'm seeing that is the same code.

Yes, credential details and link details are pretty similar but I don't want to think about the right abstraction now. Feel free to create a dedicated task so we can think about this abstraction globally, not only centered on error management.

Reply:
To be continued https://polygonid.atlassian.net/browse/PID-682